packages/fx-core/src/component/generator/utils.ts

```diff
@@ -235,3 +239,97 @@ export function zipFolder(folderPath: string): AdmZip {
   zip.addLocalFolder(folderPath);
   return zip;
 }
+
+export async function downloadDirectory(
+  sampleUrl: string,
+  dstPath: string,
+  concurrencyLimits = sampleConcurrencyLimits,
+  retryLimits = sampleDefaultRetryLimits
+): Promise<void> {
+  const urlInfo = parseSampleUrl(sampleUrl);
+  const { samplePaths, filePrefixUrl } = await getSampleFileInfo(urlInfo, retryLimits);
+  await downloadSampleFiles(
+    filePrefixUrl,
+    samplePaths,
+    dstPath,
+    urlInfo.dir,
+    retryLimits,
+    concurrencyLimits
+  );
+}
+
+type SampleUrlInfo = {
+  owner: string;
+  repository: string;
+  ref: string;
+  dir: string;
+};
+function parseSampleUrl(url: string): SampleUrlInfo {
+  const urlParserRegex = /https:\/\/github.com\/([^/]+)\/([^/]+)\/tree\/([^/]+)[/](.*)/;
+  const parsed = urlParserRegex.exec(url);
+  if (!parsed) throw new ParseUrlError(url);
+  const [owner, repository, ref, dir] = parsed.slice(1);
+  return { owner, repository, ref, dir };
+}
+
+async function getSampleFileInfo(urlInfo: SampleUrlInfo, retryLimits: number): Promise<any> {
+  const fileInfoUrl = `https://api.github.com/repos/${urlInfo.owner}/${urlInfo.repository}/git/trees/${urlInfo.ref}?recursive=1`;
+  const fileInfo = (
+    await sendRequestWithRetry(async () => {
+      return await axios.get(fileInfoUrl);
+    }, retryLimits)
+  ).data as any;
+
+  const fileInfoTree = fileInfo.tree as any[];
+  const samplePaths = fileInfoTree
+    .filter((node) => node.path.startsWith(urlInfo.dir) && node.type !== "tree")
+    .map((node) => node.path);
+  const filePrefixUrl = `https://raw.githubusercontent.com/${urlInfo.owner}/${urlInfo.repository}/${fileInfo.sha}/`;
+  return { samplePaths, filePrefixUrl };
+}
+
+async function downloadSampleFiles(
+  filePrefixUrl: string,
+  samplePaths: string[],
+  dstPath: string,
+  dir: string,
```

Comment:
The argument name is not clear enough. Please consider choosing a more descriptive name that clearly conveys the purpose of the argument.